XLS-0070d-credentials/README.md

```diff
@@ -0,0 +1,595 @@
+* The `Account` doesn't have enough reserve for the object.
+
+### 6.3. State Changes
+
+If the transaction is successful:
+* The `lsfAccepted` flag is turned on in the credential.
+* The `Credential` object is moved from the issuer's owner directory to the subject's.
+
+## 7. Transaction: `CredentialDelete`
+
+This transaction deletes a `Credential` object.
+
+It can be executed by:
+* The issuer, anytime.
+* The account, anytime.
+* Anyone, after the expiration time is up.
+
+### 7.1. Fields
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`TransactionType`| ✔️|`string`|`UInt16`|The transaction type (`CredentialDelete`).|
+|`Account`| ✔️|`string`|`AccountID`|The transaction submitter.|
+|`Subject`| |`string`|`AccountID`|The person that the credential is for. If omitted, `Account` is assumed to be the subject.|
+|`Issuer`| |`string`|`AccountID`|The issuer of the credential. If omitted, `Account` is assumed to be the issuer.|
+|`CredentialType`|✔️|`string`|`Blob`|A (hex-encoded) value to identify the type of credential from the issuer.|
+
+### 7.2. Failure Conditions
+
+* The credential described by the `Subject`, `Issuer`, and `CredentialType` fields doesn't exist.
+* The `Account` isn't the issuer or account, and the expiration hasn't passed.
+
+### 7.3. State Changes
+
+If the transaction is successful:
+* The `Credential` object is deleted.
+
+## 8. Transactions: Common Fields
+
+### 8.1. Fields
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/transactions/common-fields/) are the fields that all transactions currently have.
+
+<!--There are too many and I didn't want to list them all, it cluttered up the spec - but maybe it can be a collapsed section?-->
+
+| Field Name | Required? | JSON Type | Internal Type |
+|------------|-----------|-----------|---------------|
+|`CredentialIDs`| |`array`|`Vector256`|Credential(s) to attach to the transaction.|
+
+Transactions that accept this field:
+* Basically anything with a `Destination` field (like a payment) or something that has to be accepted by the destination (like an escrow/check).
+
+### 8.2. Failure Conditions
+
+* Any of the `CredentialIDs`:
+  * Isn't an object that exists.
+  * Isn't a `Credential` object.
+  * Is an expired `Credential` object.
+  * Has not been accepted.
+  * Isn't a credential issued to the `Account` sending the transaction.
+* `CredentialIDs` is included, but the destination doesn't have Deposit Authorization set up, or the transaction doesn't have a destination (e.g. `AccountSet`).
+* The group of `CredentialIDs` is not authorized by the destination.
+
+Note: the transaction will still fail if too many credentials are included. The exact list must be provided.
+
+### 8.3. State Changes
+
+* If the credential isn't valid, the transaction fails.
+* If the credential is expired, the credential is deleted.
+
+## 9. RPC: `deposit_authorized`
```

Comment:
Apart from the `deposit_authorized` RPC, my belief is that: no other RPC differs in the request/response format, despite the changes in the `Credential`-related content.

But certain RPCs like `account_objects` (needs to list all Credential's held by an account), `account_tx` and other `Transaction`-based RPCs have to display Credential information. Is this correct?

Reply:
Yes. That doesn't need to be specified in the spec, though, since those changes are expected (most of them don't require any code changes and will automatically work).